NAS2D/Resource/AnimationSet.cpp

```diff
@@ -126,10 +119,6 @@ namespace
 				throw std::runtime_error("Sprite version mismatch. Expected: " + std::string{SPRITE_VERSION} + " Actual: " + versionString());
 			}
 
-			// Note:
-			// Here instead of going through each element and calling a processing function to handle
-			// it, we just iterate through all nodes to find sprite sheets. This allows us to define
-			// image sheets anywhere in the sprite file.
```

Comment:
I'm a little on the fence about this comment. I'm not certain file processing code in this project is necessarily always generic enough to handle tags in any order. It may also take a more detailed analysis of the code to verify this is true here.

Reply:
It seems trying to re-write TinyXML/TinyXML2 instead of just using it as-is is causing more problems than necessary. :/ This is trivial with the original library.

Reply:
It's not the TinyXML code that's the problem. It's how it was used. Some of the code is written with positional assumptions about the XML tag layouts. Such as assuming the version string is always the first attribute. Or making assumptions about the first or only sub element of an element.